### PR TITLE
Use testcontainers for MySQL instead of mysql-connector-mxj

### PR DESCRIPTION
**Pull request by KushalP**
The mxj library is flaky and doesn't work on various platforms due to being outdated.

This PR is related to #894 and #947.

### CODE REVIEW

**Comment by KushalP** [comment]
In total, there are 14 failures as part of this branch that need figuring out.

**Comment by KushalP** [comment]
All of the failing tests:

```
org.jetbrains.exposed.sql.money.MoneyBaseTest
  testInsertSelectNull
org.jetbrains.exposed.spring.autoconfigure.ExposedAutoConfigurationTestAutoGenerateDDL
  async service should be initialized properly with completableFuture
  async service should be initialized properly
org.jetbrains.exposed.sql.tests.shared.dml.ReplaceTests
  testReplace01
org.jetbrains.exposed.sql.tests.shared.dml.JoinTests
  testJoin04
org.jetbrains.exposed.sql.tests.shared.DDLTests
  complexTest01
  testInnerJoinWithMultipleForeignKeys
  createTableWithForeignKeyToAnotherSchema
  tableExists02
  unnamedTableWithQuotesSQL
  unnamedTableWithQuotesSQLInSQLite
org.jetbrains.exposed.sql.tests.shared.SchemaTests
  create and set schema in mysql
  create existing schema and drop nonexistent schemas
  schemas exists tests
  table references table with same name in other database in mysql
org.jetbrains.exposed.sql.tests.shared.ddl.DatabaseTests
  create database test
org.jetbrains.exposed.sql.tests.shared.ddl.CreateTableTests
  test create table with same name in different schemas
```

**Comment by Tapac** [comment]
Could you please run tests with Java8 as gradle JDK?

**Comment by Tapac** [comment]
Should be fixed in master, please check.

**Comment by KushalP** [comment]
This is now ready for another review. It looks like it successfully removes mxj.

**Comment by Tapac** [comment]
@KushalP , in the current implementation you removed the possibility to run tests with `exposedDialectTestWithDocker` gradle task which uses docker-compose configs. There are to MySQL related docker-compose tests: Mysql 5.7 + jdbc 5.1 and Mysql 8.0 + jdbc 8.0.
Could you please change code to support either previous behavior, or implement similar with test-containers?

**Comment by KushalP** [comment]
@Tapac are they run in TeamCity? If possible, is it possible to get them added to TeamCity so that my PR shows up as red in this case. If I'd seen that feedback, I would have gone and fixed it. I'll work on that next.

**Comment by Tapac** [comment]
@KushalP , yes, but only master branch. You could check how to run tests locally [here](https://github.com/JetBrains/Exposed/wiki/Contributing#how-to-run-docker-based-tests) just try mysql or mysql8 and ensure what test-containers was not started.

**Comment by KushalP** [comment]
@Tapac I've tested it and it's now possible to run the `exposedDialectTestWithDocker` task for `mysql` and `mysql8` dialects.

File: exposed-tests/src/main/kotlin/org/jetbrains/exposed/sql/tests/DatabaseTestsBase.kt [hunk 1]
```diff
@@ -95,6 +90,21 @@ private val postgresSQLProcess by lazy {
...
+    SpecifiedMySQLContainer(image = "mysql:5")
+            .withDatabaseName("testdb")
+            .withEnv("MYSQL_ROOT_PASSWORD", "test")
+            .withExposedPorts().apply {
+               start()
+            }
+}
+
+private fun runTestContainersMySQL(): Boolean =
+    (System.getProperty("exposed.test.mysql.host") ?: System.getProperty("exposed.test.mysql8.host")).isBlank()
```
Comment (Tapac) [inline exposed-tests/src/main/kotlin/org/jetbrains/exposed/sql/tests/DatabaseTestsBase.kt:106]:
```suggestion
    (System.getProperty("exposed.test.mysql.host") ?: System.getProperty("exposed.test.mysql8.host")).isNullOrBlank()
```

**Pull request merged**